src/main/java/Duke.java

```diff
@@ -1,10 +1,61 @@
+import dukecommand.DukeCommand;
+import dukeexceptions.DukeException;
+import dukelist.DukeList;
+import dukeparser.DukeParser;
+import dukestorage.DukeStorage;
+import dukeui.DukeUI;
+
+import java.io.IOException;
+
+/**
+ * Represents a Duke object that runs the main function of the program
+ * Will run until user inputs 'bye' command or force quits the program.
+ */
 public class Duke {
+    private DukeList dl;
+    private DukeStorage ds;
+    private DukeUI ui;
+    private DukeParser parser;
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        Duke duke = new Duke();
+        duke.run();
+    }
+
+    /**
+     * Constructor for Duke. Each instantiation will create a new
+     * DukeStorage, DukeList, DukeUI, DukeParser for the object.
+     */
+    public Duke() {
+        try {
+            ds = new DukeStorage();
+            dl = ds.load();
+            ui = new DukeUI();
+            parser = new DukeParser();
+        } catch (IOException ioe) {
+            System.out.println(ioe.getMessage());
```

Comment:
you might need to think about better error handling to make your program work smoothly :)